instrumentation/graphql/lib/opentelemetry/instrumentation/graphql/tracers/graphql_tracer.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'opentelemetry'
+
+module OpenTelemetry
+  module Instrumentation
+    module GraphQL
+      module Tracers
+        # GraphQLTracer contains the OpenTelemetry tracer implementation compatible with
+        # the GraphQL tracer API
+        class GraphQLTracer < ::GraphQL::Tracing::PlatformTracing
+          self.platform_keys = {
+            'lex' => 'graphql.lex',
+            'parse' => 'graphql.parse',
+            'validate' => 'graphql.validate',
+            'analyze_query' => 'graphql.analyze_query',
+            'analyze_multiplex' => 'graphql.analyze_multiplex',
+            'execute_query' => 'graphql.execute_query',
+            'execute_query_lazy' => 'graphql.execute_query_lazy',
+            'execute_multiplex' => 'graphql.execute_multiplex'
+          }
+
+          def platform_trace(platform_key, key, data)
+            return yield if platform_key.nil?
+
+            tracer.in_span(platform_key || 'test', attributes: attributes_for(key, data)) do |_span|
+              yield
+            end
+          end
+
+          def platform_field_key(type, field)
+            return unless config[:enable_platform_field_key]
+
+            "#{type.graphql_name}.#{field.graphql_name}"
+          end
+
+          def platform_authorized_key(type)
+            return unless config[:enable_platform_authorized_key]
+
+            "#{type.graphql_name}.authorized"
+          end
+
+          def platform_resolve_type_key(type)
+            return unless config[:enable_platform_resolve_type_key]
```

Comment:
These config options should be documented in the README and/or with the `...::Instrumentation` class docs.

Reply:
Updated.